src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -782,9 +781,67 @@ public PreferencesViewModel(DynamoViewModel dynamoViewModel)
             var packageLoader = dynamoViewModel.Model.GetPackageManagerExtension()?.PackageLoader;            
             PackagePathsViewModel = new PackagePathViewModel(packageLoader, loadPackagesParams, customNodeManager);
 
+            PackagePathsViewModel.PackagePathOperationsEvent += PackagePathsViewModel_PackagePathOperations;
+
             this.PropertyChanged += Model_PropertyChanged;
         }
 
+        /// <summary>
+        /// Listen for changes to the custom package paths and update package paths for install accordingly
+        /// </summary>
+        private void PackagePathsViewModel_PackagePathOperations(object sender, PackagePathOperationsEventArgs e)
```

Comment:
Actually - instead of creating this new class, new event type etc - why not just watch `RootLocations` directly, it's already public and it's an observable collection, it should have events like this already I think.

Reply:
Ok, I will give it a try.